xcm-support/src/lib.rs

```diff
@@ -12,10 +12,14 @@ use sp_std::{
 	result,
 };
 
-use xcm::v0::{Error, Junction, MultiAsset, MultiLocation, Result};
+use xcm::v0::{Error, Junction, MultiAsset, MultiLocation, Result, Xcm};
 use xcm_executor::traits::{FilterAssetLocation, LocationConversion, MatchesFungible, NativeAsset, TransactAsset};
 
-use frame_support::{log, traits::Get};
+use frame_support::{dispatch::DispatchResult, log, traits::Get};
+
+pub trait XcmHandler<AccountId> {
+	fn execute_xcm(origin: AccountId, xcm: Xcm) -> DispatchResult;
```

Comment:
```suggestion
	fn execute_xcm(origin: &AccountId, xcm: Xcm) -> DispatchResult;
```

Reply:
The `xcm-handler` requires `AccountId` param.

Making it as reference would require `Clone` bound on `AccountId` generic and do clones.